exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java

```diff
@@ -38,40 +39,53 @@
  */
 public class FunTrace extends BasicFunction {
     
-    public final static FunctionSignature signature =
+    public final static FunctionSignature[] signatures = {
```

Comment:
Any interest in using the newer FunctionDSL approach? It has some advantages...

Reply:
I was in doubt to apply that pattern, but was not sure how to start....

Reply:
yes lets do that, do you have a good sample for it?

Reply:
I think the CacheModule isn't a bad example.